### PR TITLE
Integrate psp function finder

### PR DESCRIPTION
**Pull request by sozud**
Current output looks like this
```
| Ovl    | Function                 |   Length |   Branches | Jtbl   | WIP                             | %     |
|--------|--------------------------|----------|------------|--------|---------------------------------|-------|
| wrp    | DestroyCurrentEntity     |       16 |          2 |        | https://decomp.me/scratch/aKfDL | 0.99  |
| wrp    | CreateEntitiesToTheRight |       90 |         11 |        | https://decomp.me/scratch/nKKQN | 0.989 |
| wrp    | CreateEntitiesAbove      |       92 |         11 |        | https://decomp.me/scratch/ezNeo | 0.99  |
| wrp    | CreateEntitiesToTheLeft  |       97 |         12 |        |                                 |       |
| wrp    | CreateEntitiesBelow      |       99 |         12 |        |                                 |       |
| wrp    | InitRoomEntities         |      151 |         10 |        | https://decomp.me/scratch/thtl9 | 0.981 |
| wrp    | EntityEquipItemDrop      |      584 |         67 | Yes    | https://decomp.me/scratch/R4CBY | 0.993 |
| wrp    | EntityNumericDamage      |      683 |         51 |        | https://decomp.me/scratch/IyaVH | 0.989 |
| tt_000 | func_80172C30            |      731 |         65 |        |                                 |       |
| wrp    | EntityPrizeDrop          |      973 |        120 | Yes    |                                 |       |
| wrp    | EntityStageNamePopup     |      990 |         83 |        | https://decomp.me/scratch/hTUwZ | 0.972 |
| wrp    | EntityWarpRoom           |     1034 |         83 | Yes    | https://decomp.me/scratch/4MUb6 | 0.953 |
| wrp    | EntityRelicOrb           |     1186 |         87 | Yes    |                                 |       |
| tt_000 | func_80172120            |     1199 |        126 | Yes    |                                 |       |
| wrp    | EntityRedDoor            |     1248 |        112 | Yes    |                                 |       |
| wrp    | HitDetection             |     1628 |        196 |        |                                 |       |
```

I haven't tested the CI script, will just do some commits on master if it's broken

### CODE REVIEW

File: .github/workflows/validate-and-report.yml
Comment (Xeeynamo) [inline .github/workflows/validate-and-report.yml:185]: What if we change this into `gh-duplicates-${{ matrix.version }}`? We could rename later on `gh-duplicates` as `gh-duplicates-us` too and merge the two `Clone asset repository` steps

File: .github/workflows/validate-and-report.yml
Comment (Xeeynamo) [inline .github/workflows/validate-and-report.yml:219]:
```suggestion
            python3 tools/function_finder/function_finder_psp.py --use-call-trees > gh-duplicates-psp/functions.md
```
The output directory needs to match the `with > path` from the `Clone asset repository` step

**Review by Xeeynamo — APPROVED** [review]
I understand some trade-offs you made here, including using the `gh-duplicates-psp` branch and how you are searching for the `ovl_name` and `func_name`. I couldn't honestly find a better way to do this.

I am conscious of the potential spam on the #github channel in our Discord server. Adding a new branch will mean more noise. If we really decide to commit into having `gh-duplicates-*` branches we might need a AWS Lambda or something to put in the middle and filter the events. Or a Discord bot that automatically deletes any new message from the  "progress report" thing.

I left two comments. Other than that feel free to merge this any time and iterate on `master` if there are any issue. Thanks for the initiative!

**Comment by sozud** [comment]
>I am conscious of the potential spam on the #github channel in our Discord server.

How about we have a sotn-info repo or whatever and have all these autogenerated files over there?

**Comment by Xeeynamo** [comment]
> > I am conscious of the potential spam on the #github channel in our Discord server.
> 
> How about we have a sotn-info repo or whatever and have all these autogenerated files over there?

I've been thinking about that at the very beginning I introduced `gh-duplicates` and the progress page. I am not sure to remember why I decided to put everything together, I think it was due to security issues on pushing to other repos from here.

**Comment by sozud** [comment]
> I think it was due to security issues on pushing to other repos from here.

My impression it's the same as fetching the dependencies, you'd need to generate a token with push access https://stackoverflow.com/questions/64374179/how-to-push-to-another-repository-in-github-actions

From a security standpoint it seems fine if the repo is only used for these generated files, and there's a token that limits the operations to push only?

**Comment by Xeeynamo** [comment]
> > I think it was due to security issues on pushing to other repos from here.
> 
> My impression it's the same as fetching the dependencies, you'd need to generate a token with push access https://stackoverflow.com/questions/64374179/how-to-push-to-another-repository-in-github-actions
> 
> From a security standpoint it seems fine if the repo is only used for these generated files, and there's a token that limits the operations to push only?



> > I think it was due to security issues on pushing to other repos from here.
> 
> My impression it's the same as fetching the dependencies, you'd need to generate a token with push access https://stackoverflow.com/questions/64374179/how-to-push-to-another-repository-in-github-actions
> 
> From a security standpoint it seems fine if the repo is only used for these generated files, and there's a token that limits the operations to push only?

I will make some research to make a fine grain token with just write access to a specific repo. I remember it was not quite possible previously. Other than the minor github spam, I am happy to put this in until then

**Comment by joshlory** [comment]
Good to merge? I'm looking forward to this!

**Comment by Xeeynamo** [comment]
> Good to merge? I'm looking forward to this!

IMHO yes! I am waiting for @sozud to merge this and keep an eye on the CI health . I am a bit busy today but I can support you all tomorrow

**Pull request merged**
